docs/argo-server-sso.md

```diff
@@ -194,3 +194,20 @@ sso:
 # assuming customClaimGroupName: argo_groups
 workflows.argoproj.io/rbac-rule: "'argo_admins' in groups"
 ```
+
+## Filtering groups
+
+> v3.5 and above
+
+Configure the `filterGroupsRegex` field to filter the groups returned by the OIDC provider. Some use-cases for this include:
+- You have multiple applications using the same OIDC provider, and you only want to use groups that are relevant to Argo Workflows.
+- Argo Workflows uses cookies to store the groups' information. If the number of groups returned by the OIDC provider is too large, the cookie size will exceed the [4kb size limit](https://chromestatus.com/feature/4946713618939904) and the login will fail. To avoid this, use `filterGroupsRegex` to filter out the groups that are not relevant to SSO RBAC.
```

Comment:
```suggestion
- You have many groups and exceed the [4kb cookie size limit](https://chromestatus.com/feature/4946713618939904) (cookies are used to store authentication tokens). If this occurs, login will fail.
```

- follow the same sentence structure as the first bullet for grammatical parallelism.
- no need to repeat what `filterGroupsRegex` does
- cookies are very common auth mechanisms, so if you want to mention that, we can just parenthesize that
- more direct, per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)

Reply:
done